src/js_types.hpp

```diff
@@ -270,8 +276,14 @@ struct Object {
     static ValueType get_property(ContextType c, const ObjectType &o, const std::string &s) { return get_property(c, o, StringData(s)); }
     static ValueType get_property(ContextType, const ObjectType &, const String<T> &);
     static ValueType get_property(ContextType, const ObjectType &, uint32_t);
-    static void set_property(ContextType, const ObjectType &, const String<T> &, const ValueType &, PropertyAttributes attributes = None);
-    static void set_property(ContextType, const ObjectType &, uint32_t, const ValueType &);
+    static void set_property(ContextType ctx, ObjectType&& obj, const String<T>& field, const ValueType& val, PropertyAttributes attributes = None) {
```

Comment:
Explanation, since it isn't obvious what changed here or why: JSI doesn't support setting properties on const Objects, so I had to remove const here. But you can't pass a temporary to a non-const ref argument, so just doing that prevents things like `set_property(ctx, get_global(), ...)`, which we do in a few places. So I added `Object&&` overloads, that just directly call the `Object&` overload.